dali/pipeline/operators/reader/video_reader_op.cc

```diff
@@ -32,13 +32,19 @@ The video codecs can be contained in most of container file formats. FFmpeg is u
 Returns a batch of sequences of `sequence_length` frames of shape [N, F, H, W, C] (N being the batch size and F the
 number of frames).)code")
   .NumInput(0)
-  .NumOutput(1)
-  .AddArg("filenames",
+  .OutputFn([](const OpSpec &spec) {
+      std::string file_root = spec.GetArgument<std::string>("file_root");
+      return file_root.empty() ? 1 : 2;
+    })
+  .AddOptionalArg("filenames",
       R"code(File names of the video files to load.)code",
-      DALI_STRING_VEC)
+      std::vector<std::string>{})
   .AddArg("sequence_length",
       R"code(Frames to load per sequence.)code",
       DALI_INT32)
+  .AddOptionalArg("file_root",
```

Comment:
Please add info that file_root takes a precedence before filenames. Or there could be enforce that only one can be provided at the time.

Reply:
Such information is printed below and we enforce that only one of the arguments is passed. 

Do you think it's sufficient and/or whether more info should be added in docstr?

https://github.com/NVIDIA/DALI/blob/6f61997f4da71e0eaddf3c6871c9bf0ba6535c56/dali/pipeline/operators/reader/video_reader_op.h#L38-L40

Reply:
I missed that enforce. But please also add this to documentation of this argument, so user can know about it before he tries something we don't support. Like:
```
.AddOptionalArg("file_root",
R"code(Path to a directory containing data files. This option is exclusive with filenames)code",
      std::string())
```
And
```
.AddOptionalArg("filenames",
      R"code(File names of the video files to load. This option is exclusive with the file_root)code",
      DALI_STRING_VEC)
      std::vector<std::string>{})
```
```